OpenRA.Mods.Common/UpdateRules/Rules/AddCrateCollector.cs

```diff
@@ -0,0 +1,141 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OpenRA.Mods.Common.UpdateRules.Rules
+{
+	public class AddCrateCollector : UpdateRule
+	{
+		public override string Name { get { return "Add CrateCollector"; } }
+		public override string Description
+		{
+			get
+			{
+				return "Crate collection is now done via CrateCollector trait instead of crushing.";
+			}
+		}
+
+		List<string> collectingLocomotors = new List<string>();
+		List<MiniYamlNode> affectedActors = new List<MiniYamlNode>();
+
+		bool addedCrateCollector;
+
+		public override IEnumerable<string> AfterUpdate(ModData modData)
+		{
+			if (!addedCrateCollector)
+			{
+				addedCrateCollector = true;
+				foreach (var actor in affectedActors)
+					AddCrateCollectorIfNecessary(actor);
+			}
+
+			yield break;
+		}
+
+		public override IEnumerable<string> UpdateActorNode(ModData modData, MiniYamlNode actorNode)
+		{
+			var crate = actorNode.LastChildMatching("Crate");
```

Comment:
Can we please loop over ChildrenMatching? (ref the pr I reviewed earlier tonight).